.github/workflows/aem-sync-to-cloudmanager-repo.yml

```diff
@@ -0,0 +1,228 @@
+        required: true
+      cm-git-remote:
+        description: "Cloud Manager Git remote URL (without https://)"
+        required: true
+
+    outputs:
+      sync-status:
+        description: "Status of the Cloud Manager sync (success/skipped/failed)"
+        value: ${{ jobs.sync-to-cloudmanager.outputs.status }}
+
+jobs:
+  run-tests:
+    name: 🧪 Run Tests
+    runs-on: ubuntu-latest
+    if: inputs.skip-tests == false
+
+    container:
+      image: maven:${{ inputs.maven-version }}
+
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v4
+
+      - name: Cache Maven packages
+        if: ${{ !env.ACT }}  # Skip when running with act locally
+        uses: actions/cache@v4
+        with:
+          path: ~/.m2
+          key: ${{ runner.os }}-m2-${{ hashFiles('**/pom.xml') }}
+          restore-keys: ${{ runner.os }}-m2
+
+      - name: Run Maven tests
+        run: |
+          echo "🧪 Running Maven tests..."
+
+          verbose=""
+          if [ "${{ inputs.debug }}" = "true" ]; then
+            verbose="-X"
+          fi
+
+          mvn -B verify $verbose
+          echo "✅ Tests completed successfully"
+
+  build:
+    name: 🔨 Build
+    runs-on: ubuntu-latest
+    needs: [run-tests]
+    if: |
+      always() &&
+      inputs.skip-build == false &&
+      (needs.run-tests.result == 'success' || needs.run-tests.result == 'skipped')
+
+    container:
+      image: maven:${{ inputs.maven-version }}
+
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v4
+
+      - name: Cache Maven packages
+        if: ${{ !env.ACT }}  # Skip when running with act locally
+        uses: actions/cache@v4
+        with:
+          path: ~/.m2
+          key: ${{ runner.os }}-m2-${{ hashFiles('**/pom.xml') }}
+          restore-keys: ${{ runner.os }}-m2
+
+      - name: Build with Maven
+        run: |
+          echo "🔨 Building AEM project with Maven..."
+
+          verbose=""
+          if [ "${{ inputs.debug }}" = "true" ]; then
+            verbose="-X"
+          fi
+
+          mvn -B clean install $verbose
+          echo "✅ Build completed successfully"
+
+  sync-to-cloudmanager:
+    name: 📤 Sync to Cloud Manager
+    runs-on: ubuntu-latest
+    needs: [build]
+    if: |
+      always() &&
+      inputs.deploy-to-cloudmanager == true &&
+      (needs.build.result == 'success' || needs.build.result == 'skipped')
+    outputs:
+      status: ${{ steps.sync.outputs.status }}
+
+    steps:
+      - name: Checkout code with full history
+        uses: actions/checkout@v4
+        with:
+          fetch-depth: 0
+
+      - name: Configure Git
+        run: |
+          echo "⚙️ Configuring Git for Cloud Manager sync..."
+          git config --global user.name "GitHub Actions"
+          git config --global user.email "actions@github.com"
+          git config --global --add safe.directory $GITHUB_WORKSPACE
+          echo "✅ Git configuration complete"
+
+      - name: Validate required secrets
```

Comment:
This could happen in a predeploy step to verify these exist, then we can fail early.